CONTRIBUTING.md

```diff
@@ -31,123 +31,112 @@ Note that we use the Google CLA because the OpenConfig project is
 [administered and maintained by Google](https://opensource.google.com/docs/cla/#why),
 not to ascribe any specific rights to a single OpenConfig contributor.
 
-## Make a contribution
-
-To make a contribution to OpenConfig featureprofiles:
-
-1.  Open a pull request in the
-    [openconfig/featureprofiles](https://github.com/openconfig/featureprofiles)
-    repo. A brief description of the proposed addition along with references to
-    any discussion issues should be included.
-
-    *   Pull requests should be kept small. An ideal change is less than 500
-        lines. Small changes allow detailed discussions of the additions that
-        are being made to the model, whilst also ensuring that
-        course-corrections can be made early in the process. If a test is
-        growing to more than 500 lines, it may need to be broken into multiple
-        smaller tests.
-    *   New contributions should be in the feature/experimental directory.
-
-1.  The pull request should include both the suggested feature profile textproto
-    additions, as well as any relevant additions to tests. Tests should be
-    written in golang using the [ONDATRA](https://github.com/openconfig/ondatra)
-    framework.
-
-1.  The automated CI running against each pull request will check the pull
-    request for compliance. The author should resolve any issues found by CI.
-
-1.  One or more peers in the community may review the pull request.
+## Code Style
 
-1.  A feature profile repository maintainer will be reponsible for a final
-    review and approval. Only a feature repository maintainer can merge a pull
-    request to the main branch.
+All code should follow
+[Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
+and [Effective Go](https://go.dev/doc/effective_go) for writing readable Go code
+with a consistent look and feel.
 
-The aim of this process is maintain the model quality and approach that
-OpenConfig working group has strived for since its inception in 2014. Thank you
-for your contributions!
+All code and documentation should follow
+[Google developer documentation style guide](https://developers.google.com/style/word-list)
+for the use of inclusive language.
 
-## Test development style guide
+It is recommended that tests follow
+[Testing on the Toilet](https://testing.googleblog.com/) for best practices on
+design patterns.
```

Comment:
Since this is blog format, IMO it is difficult to derive any particular requirements or actions an author should perform.   Are there any specific design patterns we want to request be used?  

Perhaps another way would be to accumulate a list of recommendations.  We can start with just the obvious ones and add to them over time:

Recommended styles and design patterns used where appropriate in featureprofiles include:
- [Prefer table driven tests](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests)
- [Test clarity](https://testing.googleblog.com/2014/03/testing-on-toilet-what-makes-good-test.html)
- [Flaky test challenges](https://testing.googleblog.com/2021/03/test-flakiness-one-of-main-challenges.html)

Reply:
Listing some concrete design pattern is a great suggestion. I'll add the table driven test from the golang wiki, plus two others that I've actually referenced during PR review.